app/models/hood.rb

```diff
@@ -2,16 +2,21 @@ class Hood < ApplicationRecord
   has_many :houses
   has_many :home_owners_associations
 
+  def valuation_on_or_before(date)
+    valuation_before(date + 1.day)
+  end
+
   def valuation_before(date)
     house_count = houses.count
+    normalized_date = date.to_date
 
     # keep track of dates, and how many houses had a price at that point in time
     # default number of houses to 0
     dates_with_valuations = houses.flat_map {|house| house.price_history.keys }
 
     # array to hash
-    date_to_house_counts = dates_with_valuations.each_with_object(Hash.new(0)) do |date, date_to_occurances|
-      date_to_occurances[date] += 1
+    date_to_house_counts = dates_with_valuations.each_with_object(Hash.new(0)) do |date_with_valuation, date_to_occurances|
+      date_to_occurances[date_with_valuation] += 1
```

Comment:
What's `date_with_valuation` here?
Is the `normalized_date` coupled with the `valuation`? I.e. it's an object?
Or is this supposed to mean `valuation_date` ?

Reply:
Yeah, this can be `valuation_date`.

```
# Array[String]
dates_with_valuations = ["2021-01-02", "2021-01-03", "2021-01-01", "2021-01-02", "2021-01-02", "2021-01-03"]
```